azure-iot-device/azure/iot/device/common/errors.py

```diff
@@ -0,0 +1,151 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
```

Comment:
>-------------------------------------------------------------------------- [](start = 2, length = 74)

The errors with status codes like 4xx and 5xx seems to be more in line with http response status codes. Even if we may not use HTTP as a a protocol for some of these features , I am just trying to understand if it will be possible to use the HttpStatus class from python.

Also for the 4xx and 5xx errors I somehow feel it should be an enum or a dictionary mapping numeric status code to the error

Reply:
Great idea.  I'll do this when I start actually consuming these status codes.

---
In reply to: [302681618](https://github.com/Azure/azure-iot-sdk-python-preview/pull/136#discussion_r302681618) [](ancestors = 302681618)